.github/actions/fullnode-sync/action.yaml

```diff
@@ -0,0 +1,85 @@
+# This action runs a public fullnode using a specified branch ($BRANCH),
+# connects the public fullnode to a specified network ($NETWORK) and synchronizes the
+# node using specified syncing modes ($BOOTSTRAPPING_MODE and $CONTINUOUS_SYNCING_MODE).
+# Once the node has synced to the target version, the action completes. The node logs
+# are written to ${NODE_LOG_FILE_PATH}.
+
+# TODO: clean up the step that modifies the configs. It's a little hacky.
+runs:
+  using: composite
+  steps:
+    - uses: ./.github/actions/rust-setup
+    - name: Run dev setup and install dependencies
+      shell: bash
+      run: |
+        set -x
+        echo "${HOME}/bin/" >> $GITHUB_PATH # Default INSTALL_DIR to path
+        scripts/dev_setup.sh -b -i jq # Install dependencies
+        sudo apt-get update -y && sudo apt-get install -y expect
+    - name: Get the aptos-core branch, clone the aptos-networks repo, and get the genesis blob/waypoint
+      shell: bash
+      run: |
+        # Checkout the correct branch
+        git checkout $BRANCH
+        
+        # Clone aptos-networks and get the genesis blob and waypoint
+        cd ../
+        git clone https://github.com/aptos-labs/aptos-networks
+        cp aptos-networks/${NETWORK}/genesis.blob aptos-core/genesis.blob
+        cp aptos-networks/${NETWORK}/waypoint.txt aptos-core/waypoint.txt
+        cd aptos-core/
+    - name: Get and modify the YAML config template for the node
+      shell: bash
+      run: |
+        # Copy the node config template
+        cp config/src/config/test_data/public_full_node.yaml .
+        
+        # Update the working directory (from /opt/aptos/data to /tmp/)
+        sed -i -e 's/\/opt\/aptos\/data/\/tmp\//g' public_full_node.yaml
```

Comment:
I'm usually the worst offender of this :) but can we split most of the meat of the logic here into a separate script? That way it's easier to test and verify locally. Also gets us away from bash mess/magic